packages/flutter_tools/lib/src/base/logger.dart

```diff
@@ -395,123 +422,154 @@ enum _LogType { error, status, trace }
 /// information when stopped. When canceled, the information isn't shown. In
 /// either case, a newline is printed.
 ///
+/// The [SummaryStatus] subclass shows only a static message (without an
+/// indicator), then updates it when the operation ends.
+///
 /// Generally, consider `logger.startProgress` instead of directly creating
 /// a [Status] or one of its subclasses.
-class Status {
-  Status({this.onFinish});
+abstract class Status {
+  Status({ @required this.timeout, this.onFinish }) : assert(timeout != null);
```

Comment:
Any need to keep this public?  With the creation of `SilentStatus`, this could be private now.

Reply:
we use `Status` as a type in a lot of places.

Reply:
I'm not saying to make the type private - just the constructor

Reply:
oh I see... I don't have a strong opinion. It has to be public to subclass it outside here (e.g. in a test). Is there a reason we want to ban subclassing?

Reply:
I just tend to view overly permissive visibility the same way I view over designing - better to start out with tight visibility and open it up only when real use cases arise